extensions-core/druid-basic-security/src/main/java/org/apache/druid/security/basic/authentication/validator/LDAPCredentialsValidator.java

```diff
@@ -0,0 +1,366 @@
+        credentialVerifyDuration == null ? BasicAuthUtils.DEFAULT_CREDENTIAL_VERIFY_DURATION_SECONDS : credentialVerifyDuration,
+        credentialMaxDuration == null ? BasicAuthUtils.DEFAULT_CREDENTIAL_MAX_DURATION_SECONDS : credentialMaxDuration,
+        credentialCacheSize == null ? BasicAuthUtils.DEFAULT_CREDENTIAL_CACHE_SIZE : credentialCacheSize
+    );
+
+    this.cache = new LruBlockCache(
+        this.ldapConfig.getCredentialCacheSize(),
+        this.ldapConfig.getCredentialVerifyDuration(),
+        this.ldapConfig.getCredentialMaxDuration()
+    );
+  }
+
+  Properties bindProperties(BasicAuthLDAPConfig ldapConfig)
+  {
+    Properties properties = commonProperties(ldapConfig);
+    properties.put(Context.SECURITY_PRINCIPAL, ldapConfig.getBindUser());
+    properties.put(Context.SECURITY_CREDENTIALS, ldapConfig.getBindPassword().getPassword());
+    return properties;
+  }
+
+  Properties userProperties(BasicAuthLDAPConfig ldapConfig, LdapName userDn, char[] password)
+  {
+    Properties properties = commonProperties(ldapConfig);
+    properties.put(Context.SECURITY_PRINCIPAL, userDn.toString());
+    properties.put(Context.SECURITY_CREDENTIALS, String.valueOf(password));
+    return properties;
+  }
+
+  Properties commonProperties(BasicAuthLDAPConfig ldapConfig)
+  {
+    Properties properties = new Properties();
+    properties.put(Context.INITIAL_CONTEXT_FACTORY, "com.sun.jndi.ldap.LdapCtxFactory");
+    properties.put(Context.PROVIDER_URL, ldapConfig.getUrl());
+    properties.put(Context.SECURITY_AUTHENTICATION, "simple");
+    if (StringUtils.toLowerCase(ldapConfig.getUrl()).startsWith("ldaps://")) {
+      properties.put(Context.SECURITY_PROTOCOL, "ssl");
+      properties.put("java.naming.ldap.factory.socket", BasicSecuritySSLSocketFactory.class.getName());
+    }
+    return properties;
+  }
+
+  @Override
+  public AuthenticationResult validateCredentials(
+      String authenticatorName,
+      String authorizerName,
+      String username,
+      char[] password
+  )
+  {
+    Set<LdapName> groups;
+    LdapName userDn;
+    Map<String, Object> contexMap = new HashMap<>();
+
+    BasicAuthenticatorUserPrincipal principal = this.cache.getOrExpire(username);
+    if (principal != null && principal.hasSameCredentials(password)) {
+      contexMap.put(BasicAuthUtils.GROUPS_CONTEXT_KEY, principal.getGroups());
+      return new AuthenticationResult(username, authorizerName, authenticatorName, contexMap);
+    } else {
+      try {
+        InitialDirContext dirContext = new InitialDirContext(bindProperties(this.ldapConfig));
+
+        try {
+          SearchResult userResult = getLdapUserObject(this.ldapConfig, dirContext, username);
+          if (userResult == null) {
+            LOG.debug("User not found: %s", username);
+            return null;
+          }
+          userDn = new LdapName(userResult.getNameInNamespace());
+          groups = getGroupsFromLdap(this.ldapConfig, userResult);
```

Comment:
I think the group lookup should be moved to the `LDAPRoleProvider`, since the groups aren't needed to validate the user/password

Reply:
Done. This change required a bit more work/effort than I had expected but I agree that moving it to the LDAPRoleProvider makes more sense. Now there's no group related operations within LDAPCredentialsValidator
With this change I also had to move the groupFilters property to ldap authorizer
druid.auth.authorizer.MyBasicLDAPAuthorizer.roleProvider.groupFilters